File: icando-ae-api/internal/route/student/quiz.go
package student

import (
	"github.com/gin-gonic/gin"
	"icando/internal/handler/student"
	"icando/internal/middleware"
)

type QuizRoute struct {
	quizHandler    student.QuizHandler
	authMiddleware middleware.AuthMiddleware
}

func (r QuizRoute) Setup(group *gin.RouterGroup) {
	group = group.Group("/quiz")
	group.GET("/overview", r.quizHandler.GetQuiz)
	group.GET("/detail", r.quizHandler.GetQuizDetail)
	group.POST("/question/:id", r.quizHandler.UpdateAnswer)
	group.PATCH("/start", r.quizHandler.StartQuiz)
	group.PATCH("/submit", r.quizHandler.SubmitQuiz)
	group.GET("/review", r.quizHandler.GetQuizReview)
}

func NewQuizRoute(
	handler student.QuizHandler,
	authMiddleware *middleware.AuthMiddleware,
) *QuizRoute {
	return &QuizRoute{
		quizHandler:    handler,
		authMiddleware: *authMiddleware,
	}
}

File: icando-ae-fe/apps/assessment-web/src/services/quiz.ts
import { StudentQuiz } from "../interfaces/quiz";
import { api } from "../utils/api";

const path = "/student/quiz";

export const getQuizAvailability = async () => {
  return (await api.get(`${path}/overview`)).data.data as StudentQuiz;
};

export const getQuizDetail = async () => {
  return (await api.get(`${path}/detail`)).data.data as StudentQuiz;
};

export const updateAnswer = async (questionId: string, choiceId: number) => {
  await api.post(`${path}/question/${questionId}`, { answer_id: choiceId });
};

export const startQuiz = async () => {
  return (await api.patch(`${path}/start`)).data.data as StudentQuiz;
};

export const submitQuiz = async () => {
  await api.patch(`${path}/submit`);
};
